src/render/filters.rs

```diff
@@ -267,7 +328,23 @@ mod tests {
     use crate::template::django_rusty_templates::{EngineData, Template};
     use crate::types::{Argument, ArgumentType, Text, Variable};
 
+    use pyo3::IntoPyObjectExt;
     use pyo3::types::{PyDict, PyString};
+    static MARK_SAFE: GILOnceCell<Py<PyAny>> = GILOnceCell::new();
+
+    fn mark_safe(py: Python<'_>, string: String) -> Result<Py<PyAny>, PyErr> {
+        let mark_safe = match MARK_SAFE.get(py) {
+            Some(mark_safe) => mark_safe,
+            None => {
+                let py_mark_safe = py.import("django.utils.safestring")?;
+                let py_mark_safe = py_mark_safe.getattr("mark_safe")?;
+                MARK_SAFE.set(py, py_mark_safe.into());
+                MARK_SAFE.get(py).unwrap()
+            }
+        };
+        let safe_string = mark_safe.call1(py, (string,))?;
+        Ok(safe_string)
```

Comment:
So generally, this would the pattern to call python functions from rust?

Reply:
I think in tests its a bit overkill to use `GILOnceCell` - that's more of an optimisation.

Reply:
Fair enough :). I'll leave it as is, as all things considering it works. 

Either way, this should be good to go now!